.travis.yml:complete refactor

andy5995 opened the pull request:
When the repo is added, the public key isn't imported automatically. This removes the need to use the "allow-unauthenticated" option.

andy5995 committed:
.travis.yml:manually retrieve key, remove "allow-unauthenticated" option

andy5995 committed:
.travis.yml:add test for gcc 8

andy5995 committed:
add extra osx tests

andy5995 commented:
Added gcc 8 because Slackware current and Debian Buster use it.

andy5995 committed:
.travis.yml:specify image for osx

andy5995 commented:
hmm... testing something.....

andy5995 commented:
> hmm... testing something.....

How about another test.

andy5995 commented:
> > hmm... testing something.....
> 
> How about another test.

All right then

andy5995 committed:
complete refactor

andy5995 commented on mk/linux/setupBuildDeps.sh:
```diff
@@ -145,7 +145,7 @@ case $distribution in
 			14.04*)
 				#LTS, name > trusty, EoL April 2019
-				installcommand="apt-get install $APT_OPTIONS build-essential cmake libsdl2-dev libalut-dev libgl1-mesa-dev libglu1-mesa-dev libvorbis-dev libwxgtk3.0-dev libx11-dev liblua5.1-0-dev libjpeg-dev libpng12-dev libcurl4-gnutls-dev libxml2-dev libircclient-dev libglew-dev libftgl-dev libfribidi-dev libvlc-dev libvlccore-dev libcppunit-dev"
+				installcommand="apt-get install $APT_OPTIONS build-essential cmake libalut-dev libgl1-mesa-dev libglu1-mesa-dev libvorbis-dev libwxgtk3.0-dev libx11-dev liblua5.1-0-dev libjpeg-dev libpng12-dev libcurl4-gnutls-dev libxml2-dev libircclient-dev libglew-dev libftgl-dev libfribidi-dev libvlc-dev libvlccore-dev libcppunit-dev"
```
Are we still supporting trusty? I had to remove the libsdl2-dev dep to get the travis build to work. the dpendency won't install on trusty due to conflicts, so the source tarball is downloaded and built instead.

If we're still supporting trusty, we may need to add that to this script the way it's done in .travis.yml in the `before_install` section.

tomreyn replied:
I think we only ever meant to support OS releases during their lifespans.

Travis CI build nodes used to be running old, and default to outdated, Ubuntu releases, though. Fortunately this [does not seem to be the case now](https://docs.travis-ci.com/user/reference/overview/#virtualization-environments) (defaults to Ubuntu 16.04 LTS "Xenial", Ubuntu 18.04 LTS "Bionic" is available).

andy5995 commented on .travis.yml:
```diff
@@ -1,85 +1,115 @@
     - os: linux
+      dist: trusty
+      compiler: gcc
+      env: DEFAULT_TRAVIS_COMPILER="true" TRAVIS_DIST="trusty"
```
The default uses gcc-4. Do we want to keep a test for that or just test gcc-5-8?

tomreyn replied:
Reviewing GCC default versions currently in *supported* [Debian](https://qa.debian.org/madison.php?package=gcc&table=debian) and [Ubuntu](https://qa.debian.org/madison.php?package=gcc&table=ubuntu) releases, I think the oldest gcc version MegaGlest should currently try to support is Ubuntu 16.04 (LTS) "Xenial"'s **gcc 5.3.1** (until 04/2021, when Xenial reaches EOL).

andy5995 commented:
This is passing and ready for review. I left some line comments though with some additional questions that may need to be answered before merging.

andy5995 committed:
just use xenial, not trusty

andy5995 committed:
fail with 4.7, trying 4.8

fails building the megaglest_test target with 4.7 on xenial. was
building on trusty with 4.8 so trying that.

I think the failure is because 4.7 doesn't have some standards enabled
by default, so could be enabled by adding a flag or 2. But in this case,
just gonna try 4.8

andy5995 committed:
add std=gnu++11 to megaglest_tests build flags

andy5995 committed:
add test for embedded libs build, use BUILD_OPTIONS to reduce lines for conditions

andy5995 committed:
use trusty for gcc 4.8

andy5995 commented:
Reminder to whoever merges this.. use SQUASH & merge! ;)

andy5995 committed:
remove test on trusty with g++-4.8

andy5995 committed:
for g++-7 test, use bionic

andy5995 committed:
add test for g++-9

10 won't work yet due to issue #197

Jammyjamjamman commented:
Hey @andy5995, would it be possible to add ubuntu 18.04 and 20.04 to the tests? I think this should be good to merge, then. (It could be merged I think anyway, tbh.)

andy5995 committed:
add g++ 9 and 10 test, replace source add-on lines with single...

line to import toolchain

andy5995 commented:
> Hey @andy5995, would it be possible to add ubuntu 18.04 and 20.04 to the tests? I think this should be good to merge, then. (It could be merged I think anyway, tbh.)

Hi @Jammyjamjamman ;) Thanks for reviewing!

I rebased and added a test for g++ 10.

The test for 7, 9, and 10 will use bionic (18.04). The Ubuntu Linux build environment on Travis doesn't yet have 20.04 available for testing. https://docs.travis-ci.com/user/reference/linux/

andy5995 commented:
This is obsolete now that Github Actions is being used.
